flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -763,6 +827,7 @@ ANY, and(logical(LogicalTypeRoot.BOOLEAN), LITERAL)
     public static final BuiltInFunctionDefinition SUBSTR =
             BuiltInFunctionDefinition.newBuilder()
                     .name("substr")
+                    .callSyntax("SUBSTR", CallSyntax.SUBSTRING)
```

Comment:
Serialize into `SUBSTRING`? I don't understand why this synonym actually exists.

Reply:
Neither do I. We have those two functions in Calcite as well. We have also two separate calls in Table API.

I'd rather remove one version from Table API if we want to serialize it into the same function.